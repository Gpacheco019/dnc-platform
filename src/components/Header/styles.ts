import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  height: 6rem;
  background-color: ${({ theme }) => theme.colors.neutral.strong_gray};
  position: fixed;
`;

export const WrapperContent = styled.div`
  height: 100%;
  display: flex;
  align-items: center;
`;
