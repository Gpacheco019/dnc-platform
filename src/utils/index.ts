export * from './localStorage';
export * from './delay';
export * from './tests/__mocks__';
export { createSetupToTests } from './tests/__config__/helpers';
