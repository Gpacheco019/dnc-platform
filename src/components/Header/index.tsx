import * as S from './styles';

export type HeaderProps = {
  children: React.ReactElement;
};

const Header = ({ children }: HeaderProps) => {
  return (
    <S.Wrapper>
      <S.WrapperContent>{children}</S.WrapperContent>
    </S.Wrapper>
  );
};

export default Header;
