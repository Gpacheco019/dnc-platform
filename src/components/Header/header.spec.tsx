import { createSetupToTests } from 'utils';

import Header, { HeaderProps } from './index';

const setup = createSetupToTests<HeaderProps>(
  {
    children: (
      <>
        <p>text element</p>
      </>
    )
  },
  Header
);

describe('<Header />', () => {
  it('should be render the component', () => {
    const { getByText, container } = setup();

    expect(getByText('text element'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
