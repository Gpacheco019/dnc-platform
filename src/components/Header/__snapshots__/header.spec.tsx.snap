// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be render the component 1`] = `
.c0 {
  width: 100%;
  height: 6rem;
  background-color: #303030;
  position: fixed;
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p>
      text element
    </p>
  </div>
</div>
`;
